core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -333,8 +333,12 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
             "Kokkos::MDRangePolicy bounds error: The lower bound (" +
             std::to_string(m_lower[i]) + ") is greater than its upper bound (" +
             std::to_string(m_upper[i]) + ") in dimension " + std::to_string(i) +
-            ".";
+            ".\n";
+#if !defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
         Kokkos::abort(msg.c_str());
+#elif defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
```

Comment:
I am on the fence whether to have just 
```suggestion
#elif
```
here instead.  `Impl::log_warning()` already honors the runtime option to disable warning messages.;
Let me know what you think.

Reply:
Unless `KOKKOS_ENABLE_DEPRECATION_WARNINGS` is set to be on by default, I want to vote for changing to `#else` and always produce warning messages.
It may potentially produce noisy outputs, but it'll notify that there are definitely incorrectly set bounds for a policy somewhere in the code (which hopefully were set by mistake and are trivial to correct).

Reply:
It is `ON` by default
https://github.com/kokkos/kokkos/blob/179d2e67fd42eeb894cf9b5e976104a13de990c2/cmake/kokkos_enable_options.cmake#L52